Fix makefile to use EB environment
Simon Branford (University of Birmingham)
Based off previous patches by Ward Poelmans, Alex Domingo (Vrije Universiteit Brussel)
--- makefile.orig	2023-07-03 15:23:43.054817000 +0100
+++ makefile	2023-07-03 15:47:59.181095277 +0100
@@ -33,11 +33,11 @@
 INCLUDEDIR = /usr/local/include
 LIBDIR     = /usr/local/lib
 
-CFLAGS     ?= -O3 -Wall 
+CFLAGS     ?= $(OPTFLAGS) -Wall 
 #CFLAGS     = -g -Wall 
 
 #use this if you want only output file contain data between begin/end lines
-#CFLAGS     = -O3 -Wall -DLRS_QUIET
+#CFLAGS     = $(OPTFLAGS) -Wall -DLRS_QUIET
 
 SHLIB_CFLAGS = -fPIC
 mpicxx=mpicc
@@ -61,17 +61,17 @@
 MPLRSOBJ64=lrslong1-mplrs.o lrslib1-mplrs.o lrslibgmp-mplrs.o lrsgmp-mplrs.o lrsdriver-mplrs.o mplrs64.o
 
 lrs: ${LRSOBJ}
-	$(CC) ${CFLAGS} -DMA ${BITS} -L${LIBDIR} -o lrs ${LRSOBJ} -lgmp
-	$(CC) -O3 hvref.c -o hvref
+	$(CC) ${CFLAGS} -DMA ${BITS} $(LDFLAGS) -o lrs ${LRSOBJ} -lgmp
+	$(CC) $(OPTFLAGS) hvref.c -o hvref
 	ln -s -f lrs redund
 
 lrsMP: ${LRSOBJMP}
 	$(CC) ${CFLAGS} -DMA ${BITS} -o lrsMP ${LRSOBJMP} 
-	$(CC) -O3 hvref.c -o hvref
+	$(CC) $(OPTFLAGS) hvref.c -o hvref
 	ln -s -f lrs redund
 
 lrs64: ${LRSOBJ64}
-	$(CC) ${CFLAGS} -DMA -L${LIBDIR} -o lrs ${LRSOBJ64} -lgmp
+	$(CC) ${CFLAGS} -DMA $(LDFLAGS) -o lrs ${LRSOBJ64} -lgmp
 
 lrs.o: lrs.c
 	$(CC) ${CFLAGS} -DMA ${BITS} -c -o lrs.o lrs.c
@@ -92,22 +92,22 @@
 	$(CC) ${CFLAGS} -DMA -DSAFE ${BITS} -DLRSLONG -c -o lrslib2.o lrslib.c
 
 lrslibgmp.o: lrslib.c lrslib.h
-	$(CC) ${CFLAGS}  -DMA -DGMP -I${INCLUDEDIR} -c -o lrslibgmp.o lrslib.c
+	$(CC) ${CFLAGS}  -DMA -DGMP $(CPPFLAGS) -c -o lrslibgmp.o lrslib.c
 
 lrslibmp.o: lrslib.c lrslib.h
 	$(CC) ${CFLAGS}  -DMA -DMP -c -o lrslibmp.o lrslib.c
 
 lrsgmp.o: lrsgmp.c lrsgmp.h
-	$(CC) ${CFLAGS} -DMA -DGMP -I${INCLUDEDIR} -c -o lrsgmp.o lrsgmp.c
+	$(CC) ${CFLAGS} -DMA -DGMP $(CPPFLAGS) -c -o lrsgmp.o lrsgmp.c
 
 lrsmp.o: lrsmp.c lrsmp.h
 	$(CC) ${CFLAGS} -DMA -DMP -c -o lrsmp.o lrsmp.c
 
 inedel: inedel.c lrsgmp.h lrsgmp.c
-	$(CC) ${CFLAGS} -I${INCLUDEDIR} -L${LIBDIR} -DGMP -o inedel inedel.c lrsgmp.c -lgmp
+	$(CC) ${CFLAGS} $(CPPFLAGS) $(LDFLAGS) -DGMP -o inedel inedel.c lrsgmp.c -lgmp
 
 checkpred: checkpred.c lrsgmp.h lrsgmp.c
-	$(CC) $(CFLAGS) -I${INCLUDEDIR} -L${LIBDIR} -DGMP -o checkpred checkpred.c lrsgmp.c -lgmp
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -DGMP -o checkpred checkpred.c lrsgmp.c -lgmp
 
 lrslong1-mplrs.o: lrslong.c lrslong.h
 	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -DMA -DSAFE -DLRSLONG -DPLRS -c -o lrslong1-mplrs.o lrslong.c
@@ -122,28 +122,28 @@
 	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -DMA -DSAFE ${BITS} -DLRSLONG -DPLRS -c -o lrslib2-mplrs.o lrslib.c
 
 lrslibgmp-mplrs.o: lrslib.c lrslib.h
-	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS -I${INCLUDEDIR} -c -o lrslibgmp-mplrs.o lrslib.c
+	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS $(CPPFLAGS) -c -o lrslibgmp-mplrs.o lrslib.c
 
 lrsgmp-mplrs.o: lrsgmp.c lrsgmp.h
-	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS -I${INCLUDEDIR} -c -o lrsgmp-mplrs.o lrsgmp.c
+	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS $(CPPFLAGS) -c -o lrsgmp-mplrs.o lrsgmp.c
 
 lrsdriver-mplrs.o: lrsdriver.c lrsdriver.h lrslib.h
 	$(mpicxx) $(CFLAGS) -c -o lrsdriver-mplrs.o lrsdriver.c
 
 mplrs.o: mplrs.c mplrs.h lrslib.h lrsgmp.h
-	$(mpicxx) ${CFLAGS} -I${INCLUDEDIR} -DMA -DPLRS -DTIMES ${BITS} -DSIGNALS -D_WITH_GETLINE -c -o mplrs.o mplrs.c
+	$(mpicxx) ${CFLAGS} $(CPPFLAGS) -DMA -DPLRS -DTIMES ${BITS} -DSIGNALS -D_WITH_GETLINE -c -o mplrs.o mplrs.c
 
 mplrs64.o: mplrs.c mplrs.h lrslib.h lrsgmp.h
-	$(mpicxx) ${CFLAGS} -I${INCLUDEDIR} -DMA -DPLRS -DTIMES -DSIGNALS -D_WITH_GETLINE -c -o mplrs64.o mplrs.c
+	$(mpicxx) ${CFLAGS} $(CPPFLAGS) -DMA -DPLRS -DTIMES -DSIGNALS -D_WITH_GETLINE -c -o mplrs64.o mplrs.c
 
 mplrs: ${MPLRSOBJ} mplrsgmp
-	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DMA ${BITS} -L${LIBDIR} -o mplrs ${MPLRSOBJ} -lgmp
+	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DMA ${BITS} $(LDFLAGS) -o mplrs ${MPLRSOBJ} -lgmp
 
 mplrs64: ${MPLRSOBJ64} mplrsgmp
-	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DMA -L${LIBDIR} -o mplrs ${MPLRSOBJ64} -lgmp
+	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DMA $(LDFLAGS) -o mplrs ${MPLRSOBJ64} -lgmp
 
 mplrsgmp: mplrs.c mplrs.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c
-	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DGMP -I${INCLUDEDIR} mplrs.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR} -o mplrsgmp -lgmp
+	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DGMP $(CPPFLAGS) mplrs.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS) -o mplrsgmp -lgmp
 
 mplrs1: mplrs.c mplrs.h lrslib.c lrslib.h lrslong.c lrslong.h lrsdriver.h lrsdriver.c
 	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DPLRS -DSAFE -DLRSLONG mplrs.c lrslib.c lrslong.c lrsdriver.c -o mplrs1
@@ -158,16 +158,16 @@
 
 flint:	 	lrs.c lrslib.c lrslib.h lrsgmp.c lrsgmp.h
 		@test -d  ${INCLUDEDIR}/flint || { echo ${INCLUDEDIR}/flint not found; exit 1; }
-		$(CC) -O3 -DFLINT -I/usr/local/include/flint lrs.c lrslib.c lrsgmp.c lrsdriver.c -L/usr/local/lib -Wl,-rpath=/usr/local/lib -lflint -o lrsflint -lgmp
-#		$(CC) -O3 -DFLINT -I${INCLUDEDIR} -I${INCLUDEDIR}/flint lrs.c lrsdriver.c lrslib.c lrsgmp.c -L${LIBDIR} -lflint -o lrsflint -lgmp
+		$(CC) $(OPTFLAGS) -DFLINT -I/usr/local/include/flint lrs.c lrslib.c lrsgmp.c lrsdriver.c -L/usr/local/lib -Wl,-rpath=/usr/local/lib -lflint -o lrsflint -lgmp
+#		$(CC) $(OPTFLAGS) -DFLINT $(CPPFLAGS) $(CPPFLAGS)/flint lrs.c lrsdriver.c lrslib.c lrsgmp.c $(LDFLAGS) -lflint -o lrsflint -lgmp
 
 mplrsflint:	mplrs.c mplrs.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.c lrsdriver.h
-	${mpicxx} ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DFLINT -I${INCLUDEDIR}/flint -DPLRS -o mplrsflint mplrs.c lrsdriver.c lrslib.c lrsgmp.c -L${LIBDIR} -lflint -lgmp
+	${mpicxx} ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DFLINT $(CPPFLAGS)/flint -DPLRS -o mplrsflint mplrs.c lrsdriver.c lrslib.c lrsgmp.c $(LDFLAGS) -lflint -lgmp
 
 #comment out lines with ${BITS} if __int128 not supported by your C compiler
 
 lrsgmp:		lrs.c lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c 
-		$(CC)  ${CFLAGS}  -DGMP -I${INCLUDEDIR} -o lrsgmp lrs.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR}  -lgmp
+		$(CC)  ${CFLAGS}  -DGMP $(CPPFLAGS) -o lrsgmp lrs.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS)  -lgmp
 		ln -s -f lrsgmp redundgmp
 
 single:		lrs.c lrslong.c lrslong.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c
@@ -179,32 +179,32 @@
 		ln -s -f lrs2 redund2
 
 allmp:		lrs.c lrslib.c lrslib.h lrsmp.c lrsmp.h lrsdriver.h lrsdriver.c
-		$(CC) -Wall -O3 -DMP  -o lrsmp lrs.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -Wall -O3  -DSAFE -DLRSLONG -o lrs1 lrs.c lrslib.c lrsdriver.c lrslong.c
-		$(CC) -Wall -O3  -DSAFE -DLRSLONG ${BITS} -o lrs2 lrs.c lrslib.c lrsdriver.c lrslong.c
-		$(CC) -O3 -DMP -DLRS_QUIET   -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsdriver.c lrsmp.c -static
-		$(CC) -O3 -DMP -o setupnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3 -DMP -o setupnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3  -o 2nash 2nash.c
+		$(CC) -Wall $(OPTFLAGS) -DMP  -o lrsmp lrs.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) -Wall $(OPTFLAGS)  -DSAFE -DLRSLONG -o lrs1 lrs.c lrslib.c lrsdriver.c lrslong.c
+		$(CC) -Wall $(OPTFLAGS)  -DSAFE -DLRSLONG ${BITS} -o lrs2 lrs.c lrslib.c lrsdriver.c lrslong.c
+		$(CC) $(OPTFLAGS) -DMP -DLRS_QUIET   -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsdriver.c lrsmp.c -static
+		$(CC) $(OPTFLAGS) -DMP -o setupnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS) -DMP -o setupnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS)  -o 2nash 2nash.c
 
 demo:	lpdemo1.c lrslib.c lrsdriver.c lrslib.h lrsgmp.c lrsgmp.h
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lpdemo1 lpdemo1.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lpdemo lpdemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lpdemo2 lpdemo2.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o vedemo  vedemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o chdemo  chdemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lpdemo1 lpdemo1.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lpdemo lpdemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lpdemo2 lpdemo2.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o vedemo  vedemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o chdemo  chdemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
 
 lrsnash:	lrsnash.c lrsnashlib.c lrslib.c lrsnashlib.h lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
 
 alllrsnash:	lrsnash.c nashdemo.c lrsnashlib.c lrslib.c lrsnashlib.h lrslib.h lrsgmp.c lrsgmp.h lrslong.h lrsdriver.h lrsdriver.c
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnashgmp lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash1 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash2 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE ${BITS}
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o nashdemo nashdemo.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c -lgmp -DGMP
-		$(CC) -O3 -DMP -o setupnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3 -DMP -o setupnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3  -I${INCLUDEDIR} -L${LIBDIR} -o 2nash 2nash.c
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnashgmp lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnash1 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnash2 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE ${BITS}
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o nashdemo nashdemo.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c -lgmp -DGMP
+		$(CC) $(OPTFLAGS) -DMP -o setupnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS) -DMP -o setupnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS)  $(CPPFLAGS) $(LDFLAGS) -o 2nash 2nash.c
 		cp lrsnashgmp lrsnash
 
 ######################################################################
@@ -226,7 +226,7 @@
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o \
 	${SHLIBOBJ2}
 
-SHLIBBIN=lrs-shared lrsnash-shared
+SHLIBBIN=lrs-shared lrsnash-shared lrsgmp mplrs mplrsgmp
 
 # Building (linking) the shared library, and relevant symlinks.
 
@@ -248,12 +248,12 @@
 
 
 lrsnash-shared: ${SHLINK}  lrsnash.c
-	$(CC) ${CFLAGS} -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp
+	$(CC) ${CFLAGS} -DGMP -DMA lrsnash.c  lrsnashlib.c $(CPPFLAGS) -o $@ -L . -llrs -lgmp
 
 # driver object files
 
 lrs-shared.o: lrs.c
-	$(CC) ${CFLAGS} -DMA ${BITS} -L${LIBDIR} -c -o $@ lrs.c
+	$(CC) ${CFLAGS} -DMA ${BITS} $(LDFLAGS) -c -o $@ lrs.c
 
 # build object files for the shared library
 
@@ -291,6 +291,8 @@
 	install -t $(DESTDIR)${prefix}/lib $(SHLIB)
 	cd $(DESTDIR)${prefix}/lib && ln -sf $(SHLIB) $(SHLINK)
 	cd $(DESTDIR)${prefix}/lib && ln -sf $(SHLIB) $(SONAME)
+	cd $(DESTDIR)${prefix}/bin && ln -sf lrs redund
+	cd $(DESTDIR)${prefix}/bin && ln -sf lrsgmp redundgmp
 
 install-common:
 	mkdir -p $(DESTDIR)${prefix}/include/lrslib
